Spigot-API-Patches/0243-Added-EntityItemDamageEvent.patch

```diff
@@ -0,0 +1,91 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Tue, 22 Dec 2020 13:51:06 -0800
+Subject: [PATCH] Added EntityItemDamageEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/EntityItemDamageEvent.java b/src/main/java/io/papermc/paper/event/entity/EntityItemDamageEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ee4090263f14a5d0f9606e49e36e73908528b13a
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/EntityItemDamageEvent.java
+@@ -0,0 +1,79 @@
++package io.papermc.paper.event.entity;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when an item on or used by an entity takes durability damage as a result of being hit/used.
++ *
++ * <p>
++ *     NOTE: default vanilla behaviour dictates that armor/tools picked up by
++ *     mobs do not take damage (except via Thorns).
++ * </p>
++ */
++public class EntityItemDamageEvent extends EntityEvent implements Cancellable {
```

Comment:
```suggestion
+public class EntityDamageItemEvent extends EntityEvent implements Cancellable {
```
?

Reply:
Doesn't that sound like its fired when an entity damages an item? Which isn't really what this is.

Reply:
EntityItemDamageEvent makes it sound like I can cancel the damage to Item entities (like throwing an item into fire)

Reply:
There already is a PlayerItemDamageEvent which is called when this stuff happens to a player.

Reply:
@BillyGalbreath's comment is why I suggested this change.

Reply:
But then the naming isn't consistent with the already existing player event?

Reply:
Names not being consistent does not worry me.

Reply:
Ok, changed the name